libstuff/STCPManager.cpp

```diff
@@ -228,8 +206,45 @@ STCPManager::Socket::Socket(int sock, STCPManager::Socket::State state_, SX509*
     sentBytes(0), recvBytes(0)
 { }
 
+STCPManager::Socket::Socket(const string& host, SX509* x509)
```

Comment:
This constructor does what `openSocket` used to do, but now in a place that makes it easy to reason about what does and doesn't set Socket internal state.